### PR TITLE
Add colorFilterLayer/Widget

### LINKED ISSUE #5692: Add a ColorFilter widget
I want to be able to turn a given widget into a certain color similar to how you can do this with Images.

### PR DESCRIPTION
**Pull request by dnfield**
## Description

Adds a ColorFilter layer and widget.  customer: octopod would like to use this to preview color blindness filters on widget trees.

Depends on https://github.com/flutter/engine/pull/9641 - tests will fail until that rolls in.

## Related Issues

Fixes #5692 

## Tests

I added the following tests:

Test that ColorFilterLayer does the right thing when it changes.

Golden tests for ColorFiltered widget.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change (though the upstream engine change is).

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/rendering/object.dart [hunk 1]
```diff
@@ -444,6 +444,23 @@ class PaintingContext extends ClipContext {
     }
   }
 
+  /// Blend further painting with a color filter.
+  /// * `offset` is the offset from the origin of the canvas' coordinate system
```
Comment (collinjackson) [inline packages/flutter/lib/src/rendering/object.dart]: Need an extra line here

```suggestion
  ///
  /// * `offset` is the offset from the origin of the canvas' coordinate system
```

**Review by collinjackson — APPROVED** [review]
lgtm modulo minor documentation nit

**Comment by dnfield** [comment]
Oh man I could have sworn the build was green when I merged this.  Whups.

**Pull request merged**